docs/rfcs/0561-list-with-metadata.md

```diff
@@ -0,0 +1,72 @@
+- Proposal Name: `list_with_metadata`
+- Start Date: 2022-08-23
+- RFC PR: None
+- Tracking Issue: None
+
+# Summary
+
+Add a `metadata` field for `DirEntry`s.
+
+```rust
+pub struct DirEntry {
+    acc: Arc<dyn Accessor>,
+
+    metadata: ObjectMetadata,
+    path: String,
+}
+
+impl DirEntry {
+    // get size of file
+    pub fn len(&self) -> u64 {
```

Comment:
I prefer to return `Option<u64>` instead:

- If `is_some()`, user can use this value directly.
- If `is_none`, users can try fetch this meta by `de.metadata().await`

Reply:
> And for this proposal's naming, `list_with_metadata` doesn't reflect the thing we want to do. How about `reuse_metadata_during_list`?

I renamed it to `list_metadata_reuse`